hack/e2e/utils.sh

```diff
@@ -52,3 +52,31 @@ kill_podman() {
         fi
     fi
 }
+
+setup_environment() {
+    echo "########## 🌐 Setting up Azure account and secrets ##########"
+    az account set -s "$AZURE_SUBSCRIPTION_ID"
+    SECRET_SA_ACCOUNT_NAME="$SECRET_SA_ACCOUNT_NAME" make secrets
+    . secrets/env
+    export CI=true
+    echo "Environment setup complete."
+}
+
+install_docker_dependencies() {
+    echo "########## 🐳 Installing Docker and Docker Compose Plugin ##########"
+    sudo apt-get update
+    sudo apt-get install -y ca-certificates curl gnupg
+    sudo install -m 0755 -d /etc/apt/keyrings
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo tee /etc/apt/keyrings/docker.asc
+    sudo chmod a+r /etc/apt/keyrings/docker.asc
+    echo \
+    "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+    $(. /etc/os-release && echo \"$VERSION_CODENAME\") stable" | \
+    sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+    sudo apt-get update
+    sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin make
```

Comment:
if this is running in a pipeline, we will want to pin each of these versions to prevent instability in the pipeline executions. I can't speak for GH actions, but ADO has a specific task for pinning a particular docker version which would reduce our code maintenance on this: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/docker-installer-v0?view=azure-pipelines